core/src/main/java/org/infinispan/xsite/irac/IracManagerKeyChangedState.java

```diff
@@ -92,6 +103,27 @@ public void discard() {
       STATUS_UPDATER.lazySet(this, Status.DONE);
    }
 
+   @Override
+   public void successFor(XSiteBackup site) {
+      synchronized (sentTo) {
+         sentTo.add(site);
+      }
+   }
+
+   @Override
+   public boolean wasSuccessful(XSiteBackup site) {
+      synchronized (sentTo) {
+         return sentTo.contains(site);
+      }
+   }
+
+   @Override
+   public Collection<XSiteBackup> successfullySent() {
```

Comment:
Only used to check for contains all. `containsAll` can be invoked directly here and avoid copying.